cpp/src/io/parquet/page_enc.cu

```diff
@@ -131,38 +166,45 @@ __global__ void __launch_bounds__(512) gpuInitPageFragments(PageFragment *frag,
     dtype_len_in = (dtype == BYTE_ARRAY) ? sizeof(nvstrdesc_s) : dtype_len;
   }
   __syncthreads();
-  nrows = s->frag.num_rows;
-  for (uint32_t i = 0; i < nrows; i += 512) {
+
+  nvals           = s->frag.num_leaf_values;
+  start_value_idx = s->start_value_idx;
+
+  for (uint32_t i = 0; i < nvals; i += 512) {
```

Comment:
Make this a constexpr somewhere and document what it means.  512 is a magic number otherwise.